### PR TITLE
modify delay generator for COSMIC beamline

### PR DESCRIPTION
**Pull request by JulReinhardt**
started to modify fastccd.py to support delay generator at COSMIC beamline. Not confident that any of this makes sense...

### CODE REVIEW

File: xicam/Acquire/devices/fastccd.py [hunk 1]
```diff
@@ -336,33 +337,29 @@ def __init__(self, *args, **kwargs):
...
-            self.parent.dg1.G.set(gg)
-            self.parent.dg1.H.set(hh)
-            self.parent.dg2.A.set(0)
-            self.parent.dg2.B.set(0.0005)
+            self.parent.dg1.trigger_rate.set(tr)
+            self.parent.dg1.trigger_on_off.set(tm)
+            self.parent.dg1.delay_time.set(dt)
+            self.parent.dg1.shutter_time.set(st)
+            self.parent.dg1.init.set()
+            self.parent.dg1.reset.set()
```
Comment (ronpandolfi) [inline xicam/Acquire/devices/fastccd.py]: Is it right to initialize and then immediately reset? What would this mean for the delay generator? Also, can you `.set` without args?

File: xicam/Acquire/devices/fastccd.py [hunk 2]
```diff
@@ -385,8 +382,33 @@ def get(self):
...
+#     def __init__(self, prefix, **kwargs):
+#         super().__init__(prefix + '-RB', write_pv=prefix + '-SP', **kwargs)
+
+
+class DelayGenerator(Device):
+    trigger_rate = Cpt(EpicsSignalWithRBV, '.trigger_rate')
+    trigger_on_off = Cpt(EpicsSignalWithRBV, '.trigger_on_off')
+    delay_time = Cpt(EpicsSignalWithRBV, '.delay_time')
+    shutter_time = Cpt(EpicsSignalWithRBV, '.shutter_time')
+    init = Cpt(EpicsSignal, '.init')
```
Comment (ronpandolfi) [inline xicam/Acquire/devices/fastccd.py]: "init"->"initialize"

File: xicam/Acquire/devices/fastccd.py [hunk 3]
```diff
@@ -10,7 +10,6 @@
 from ophyd import Component as Cpt
 from ophyd.device import FormattedComponent as FCpt
 from ophyd import AreaDetector
-from ophyd.utils import set_and_wait
```
Comment (ronpandolfi) [inline xicam/Acquire/devices/fastccd.py:13]: `set_and_wait` is still needed. Why was it removed?

File: xicam/Acquire/devices/fastccd.py [hunk 4]
```diff
@@ -336,33 +337,29 @@ def __init__(self, *args, **kwargs):
         self._readout = 0.080
         super().__init__(*args, **kwargs)
 
-    def set(self, exp):
+    def set(self, exp: float, triggerrate: float = 1., trigger_on: bool = 1, delaytime: float=1e-3, shuttertime: float=140e-3):
```
Comment (ronpandolfi) [inline xicam/Acquire/devices/fastccd.py]: use booleans for boolean values

File: xicam/Acquire/devices/fastccd.py (hunk 3 above)
Comment (JulReinhardt) [reply xicam/Acquire/devices/fastccd.py:13]: good question, by accident, can't explain when that happened

File: xicam/Acquire/devices/fastccd.py (hunk 1 above)
Comment (JulReinhardt) [reply xicam/Acquire/devices/fastccd.py]: ah true, that is useless. Did not think it through.

File: xicam/Acquire/devices/fastccd.py [hunk 5]
```diff
@@ -221,10 +220,34 @@ class FCCDCam(AreaDetectorCam):
...
+        self.exposure_time.sim_set_putter(self.exposure_time_putter)
+
+    def exposure_time_putter(self, value):
+        acquire_time = value + self.shutter_close_delay.get() + self.shutter_open_delay.get()
+        acquire_period = acquire_time + self.readout_time.get()
+
+        self.parent.dg1.init.set(1)
+        self.acquire_time.set_and_wait(acquire_time)
+        self.acquire_period.set_and_wait(acquire_period)
+        self.parent.dg1.trigger_rate.set_and_wait(acquire_period)
```
Comment (JulReinhardt) [inline xicam/Acquire/devices/fastccd.py]: trigger rate should be in Hz --> 1/acquire_period ?

**Comment by ronpandolfi** [comment]
(effectively merged without PR)

**Pull request closed without merging**